src/deploy/functions/build.ts

```diff
@@ -331,6 +331,8 @@
       }
       if (bdEndpoint.serviceAccount) {
         bkEndpoint.serviceAccountEmail = bdEndpoint.serviceAccount;
+      } else {
+        bkEndpoint.serviceAccountEmail = undefined;
```

Comment:
It's not _technically_ the same, since I don't think setting something undefined or null are equivalent (at least, it breaks my test cases), but it should be the same as far as the CF3 backend is concerned. That said, I'm having some difficulty getting renameIfPresent to work in the test cases.

Reply:
Specifically, when testing the back-converted Backend object:
```
      AssertionError: expected { Object (requiredAPIs, endpoints, ...) } to deeply equal { Object (requiredAPIs, endpoints, ...) }
      + expected - actual

               "platform": "gcfv2"
               "project": "project"
               "region": "region"
               "runtime": "node14"
      -        "serviceAccountEmail": [null]
               "timeoutSeconds": 60
             }
           }
         }

```

Which is wild since a Backend can't even _have_ null as the value of serviceAccountEmail, I think?

Reply:
If I can frustrate you even more, there's a difference between setting it to undefined and simply not setting it.

```
> const obj = {}
undefined
> "a" in obj
false
> obj.a
undefined
> obj.a = undefined
undefined
> "a" in obj
true
> obj.a
undefined
> delete obj.a
true
> "a" in obj
false
```

Is it wrong to use `renameIfPresent` for now which will leave the field missing instead of undefined? I also need to do a really scary change where `renameIfPresent` actually copies nulls so we can unbreak the container contract, but I'll do that later.